### PR TITLE
interpret radio commands

### PR DESCRIPTION
**Pull request by bbrk24**
diff view isn't terribly helpful because I extracted existing code to new functions

### CODE REVIEW

File: sketch_oct9a.ino [hunk 1]
```diff
@@ -176,11 +147,19 @@ void loop() {
...
+void readGPS() {
+  if (gps.getStatus() != GPS::ACTIVE) {
+    gps.initialize();
+    updateGPSLEDs();
+  }
+
+  gps.getLocation(&last_coords);
+}
+
+void readOtherSensors() {
```
Comment (DarthSethious1) [inline sketch_oct9a.ino]: Still call this 'readAltIMU', not 'readOtherSensors'

File: sketch_oct9a.ino [hunk 2]
```diff
@@ -201,7 +180,98 @@ void loop() {
...
+    last_alt
+#if CAPSULE == 2
+    ,
+    last_voc,
+    last_humid,
+    last_temp
+#endif
+  );
+
+  // In theory, this should send the data to the radio.
```
Comment (DarthSethious1) [inline sketch_oct9a.ino]: Thank you for the comment

File: sketch_oct9a.ino [hunk 3]
```diff
@@ -201,7 +180,98 @@ void loop() {
...
+  // In practice, this is completely untested. If nothing is connected to the UART pins, however,
+  // this doesn't do anything, so it's harmless to leave in for now.
+  Serial1.write(
+    (char*)&f,
+    FRAME_SIZE()
+  );
+}
+
+void setup() {
+  Serial1.begin(230400);
```
Comment (DarthSethious1) [inline sketch_oct9a.ino]: Put a comment in the code on this line that this is the actual rate we will be sending at. You could even do this in a 'DEFINE' statement closer to the top of the file.

File: sketch_oct9a.ino [hunk 4]
```diff
@@ -201,7 +180,98 @@ void loop() {
...
+  digitalWrite(5, LOW);
+  digitalWrite(6, LOW);
+  digitalWrite(7, LOW);
+
+  // Wait for a command from the radio. Sometimes the sensors need multiple tries, and calling
+  // initialize too many times does nothing, so put the initialize call in the loop.
+  do {
+    initializeAll();
+
+    if (Serial1.available()) {
```
Comment (DarthSethious1) [inline sketch_oct9a.ino:244]: why is it 'Serial1' ? There doesn't seem to be any 'Serial0' or 'Serial2'. I'm assuming this serial object is dealing with the radio/capsule interface, or is it some other interface?

File: sketch_oct9a.ino [hunk 5]
```diff
@@ -201,7 +180,98 @@ void loop() {
...
+      //The string returned from readStringUntil does not include the terminator
+      String command = Serial1.readStringUntil('\n');
+
+      if (command == "wake") {
+        break;
+      } else if (command == "status") {
```
Comment (DarthSethious1) [inline sketch_oct9a.ino]: the command name 'status' kinda makes me think of a sanity check, and not necessarily actually getting data. If you guys agree, do we want to change the name of the command to 'getInfo' or something more like that?

File: sketch_oct9a.ino [hunk 6]
```diff
@@ -227,40 +297,15 @@ void loop() {
...
-  // In practice, this is completely untested. If nothing is connected to the UART pins, however,
-  // this doesn't do anything, so it's harmless to leave in for now.
-  Serial1.write(
-    (char*)&f,
-    FRAME_SIZE()
-  );
+  sendDataToRadio();
 }
 
 #endif
```
Comment (DarthSethious1) [inline sketch_oct9a.ino:314]: As a whole, I'm still not 100% sure how the code is multi-threaded. I see a few different loops going on, but next time we meet Riley I would like for you to take a few minutes and explain it to me again. Hopefully it'll stick this time.

File: sketch_oct9a.ino (hunk 4 above)
Comment (bbrk24) [reply sketch_oct9a.ino:244]: On the MKR Zero specifically, `Serial` is the USB port and `Serial1` is the UART pins.

File: sketch_oct9a.ino (hunk 6 above)
Comment (bbrk24) [reply sketch_oct9a.ino:314]: Discussed in person.

File: sketch_oct9a.ino (hunk 5 above)
Comment (bbrk24) [reply sketch_oct9a.ino]: These command names were Carson's idea. We should discuss this in the Discord.

**Comment by bbrk24** [comment]
Even though we haven't tested sending commands to the Arduino, I'm marking this as ready to move it along. There are other things I want to implement (like stopping data collection when landing) that require this.

**Review by CClarke513 — APPROVED** [review]
Fine by me - we know we can send data on our end

**Review by DarthSethious1 — APPROVED** [review]
Looks good, mostly just shuffling around some code, not any major rewriting except for the sync word.

**Pull request merged**